modules/apps/account/account-service/src/main/java/com/liferay/account/internal/retriever/AccountUserRetrieverImpl.java

```diff
@@ -40,9 +60,136 @@
 				accountEntryUserRel.getAccountUserId()));
 	}
 
+	@Override
+	public List<User> searchAccountUsers(
+			long accountEntryId, String keywords, int status, int cur,
+			int delta, String sortField, boolean reverse)
+		throws PortalException {
+
+		SearchResponse searchResponse = _getSearchResponse(
+			accountEntryId, keywords, status, cur, delta, sortField, reverse);
+
+		SearchHits searchHits = searchResponse.getSearchHits();
+
+		return TransformUtil.transform(
+			searchHits.getSearchHits(),
+			searchHit -> {
+				Document document = searchHit.getDocument();
+
+				long userId = document.getLong("userId");
+
+				return _userLocalService.getUser(userId);
+			});
+	}
+
+	@Override
+	public long searchCountAccountUsers(
+			long accountEntryId, String keywords, int status)
+		throws PortalException {
+
+		SearchResponse searchResponse = _getSearchResponse(
+			accountEntryId, keywords, status, QueryUtil.ALL_POS,
+			QueryUtil.ALL_POS, null, false);
+
+		return searchResponse.getTotalHits();
+	}
+
+	private SearchResponse _getSearchResponse(
+			long accountEntryId, String keywords, int status, int cur,
+			int delta, String sortFieldName, boolean reverse)
+		throws PortalException {
+
+		AccountEntry accountEntry = _accountEntryLocalService.getAccountEntry(
+			accountEntryId);
+
+		SearchRequestBuilder searchRequestBuilder =
+			_searchRequestBuilderFactory.builder();
+
+		searchRequestBuilder.entryClassNames(
+			User.class.getName()
+		).withSearchContext(
+			searchContext -> {
+				boolean andSearch = false;
+
+				if (Validator.isNull(keywords)) {
+					andSearch = true;
+				}
+				else {
+					searchContext.setKeywords(keywords);
+				}
+
+				searchContext.setAndSearch(andSearch);
+
+				Map<String, Serializable> attributes = new HashMap<>();
+
+				attributes.put("city", keywords);
+				attributes.put("country", keywords);
+				attributes.put("emailAddress", keywords);
+				attributes.put("firstName", keywords);
+				attributes.put("fullName", keywords);
+				attributes.put("lastName", keywords);
+				attributes.put("middleName", keywords);
+				attributes.put("region", keywords);
+				attributes.put("screenName", keywords);
+				attributes.put("street", keywords);
+				attributes.put("zip", keywords);
+
+				attributes.put(
+					"accountEntryIds",
+					new long[] {accountEntry.getAccountEntryId()});
+				attributes.put("params", new LinkedHashMap<>());
+				attributes.put("status", status);
+
+				searchContext.setAttributes(attributes);
+
+				searchContext.setCompanyId(accountEntry.getCompanyId());
```

Comment:
Extracting this to a helper method might help with readability and follow the pattern in UserLocalServiceImpl.buildSearchContext